ext/standard/head.c

```diff
@@ -121,6 +120,9 @@ PHPAPI int php_setcookie(zend_string *name, zend_string *value, time_t expires,
 	if (domain) {
 		len += ZSTR_LEN(domain);
 	}
+	if (samesite) {
+		len += ZSTR_LEN(samesite);
+	}
 
 	cookie = emalloc(len + 100);
```

Comment:
This is smelly! Apparently, we're not calculating the correct length. If 100 additional bytes (why 100?) won't suffice, we'd have problem. Unless we're able to calculate the exact length (or to show that 100 is always sufficient), refactoring to the smart_str or smart_string API might be sensible.